### PR TITLE
Exclude spec files from AoT

### LINKED ISSUE #2736: AoT still compile spec files
> ## Please provide us with the following information:
### OS?

> Linux, Ubuntu
### Versions.

> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:
> 
> ```
> angular-cli: 1.0.0-beta.17
> node: 6.7.0
> os: linux x64
> ```
### Repro steps.

https://github.com/angular/angular/issues/11276
1. Generate new project

```
ng new test-app
```
1. Open app.component.spec.ts
   and add RouterTestingModule

``` ts
import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

describe('App: TestBug', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent
      ],
      imports: [
        RouterTestingModule
      ]
    });
  });
```
1. Then try build

```
ng build -prod -aot
```
1. Breaks

``` sh
$ ng build -prod -aot                                                                              
Could not start watchman; falling back to NodeWatcher for file system events.                      
Visit http://ember-cli.com/user-guide/#watchman for more info.                                     
  0% compilingparameters received {"filePath":"/home/antony-budianto/Desktop/personal-project/test-
bug/node_modules/@angular/common/src/location/location.d.ts","name":"Location"} which is not a Stat
icSymbol                                                                                           
Error: parameters received {"filePath":"/home/antony-budianto/Desktop/personal-project/test-bug/nod
e_modules/@angular/common/src/location/location.d.ts","name":"Location"} which is not a StaticSymbo
l 
```

I know the issue about symbol already fixed in `2.1.0`, but cli still compile spec files

Current workaround:
- Add `**/*.spec.ts` to `exclude` option in `src/tsconfig.json`
### The log given by the failure.

Please see https://github.com/angular/angular/issues/11276
### Mention any other details that might be useful.

AoT should not compile spec files

> ---
> 
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by antonybudianto**
fix #2736

### CODE REVIEW

File: packages/webpack/src/plugin.ts [hunk 1]
```diff
@@ -88,7 +88,8 @@ export class AotPlugin {
     }
 
     const tsConfig = tsc.readConfiguration(options.tsConfigPath, basePath);
-    this._rootFilePath = tsConfig.parsed.fileNames;
+    this._rootFilePath = tsConfig.parsed.fileNames
+      .filter(fileName => fileName.indexOf('.spec.ts') < 0);
```
Comment (mxii) [inline packages/webpack/src/plugin.ts]: to be 100% sure: `.filter(fileName => fileName.match(/.spec.ts$/)  === null);` ? :)

Comment (antonybudianto) [inline packages/webpack/src/plugin.ts]: i like` /\.spec\.ts$/.test(fileName)` better since it return boolean directly

**Comment by filipesilva** [comment]
Tests were failing but it wasn't related to this PR, and fixed in https://github.com/angular/angular-cli/pull/2832.

While I understand that this fix works, it shouldn't be necessary because of https://github.com/angular/angular-cli/blob/master/packages/angular-cli/models/webpack-build-typescript.ts#L57. Spec files shouldn't even be included. @hansl can you have a look?

**Comment by antonybudianto** [comment]
@filipesilva , https://github.com/angular/angular-cli/blob/master/packages/angular-cli/models/webpack-build-typescript.ts#L57 does exclude the file from what module will **imports**, but not from ngc/tsc, since in fact spec files still get compiled by ngc

**Comment by antonybudianto** [comment]
@filipesilva @hansl , any updates on this?

**Comment by hansl** [comment]
LGTM.

**Comment by ptomaszi** [comment]
When can we have that fix available through NPM?

**Comment by antonybudianto** [comment]
@ptomaszi It's on npm since beta 20-1

**Comment by BrainCrumbz** [comment]
@antonybudianto somehow off-topic, but how can one "log the fileNames" that gest compiled by ngc? Thanks

**Pull request merged**